docs/dev/stream/state/state.md

```diff
@@ -266,6 +266,92 @@ a `ValueState`. Once the count reaches 2 it will emit the average and clear the
 we start over from `0`. Note that this would keep a different state value for each different input
 key if we had tuples with different values in the first field.
 
+### State time-to-live (TTL)
+
+A time-to-live (TTL) can be assigned to the keyed state value. 
+In this case it will expire after the configured TTL
+and its stored value will be cleaned up based on the best effort.
+Depending on configuration, the expired state can become unavailable for read access
+even if it is not cleaned up yet. In this case it behaves as if it does not exist any more.
+
+The collection types of state support TTL on entry level: 
+separate list elements and map entries expire independently. 
```

Comment:
The state collection types support per-entry TTLs: list elements and map entries expire independently.